pkg/functions/client.go

```diff
@@ -737,6 +760,30 @@ func (c *Client) Deploy(ctx context.Context, f Function, opts ...DeployOption) (
 		return f, err
 	}
 
+	// If Redeployment to NEW namespace was successful -- undeploy dangling Function in old namespace.
+	// On forced namespace change (using --namespace flag)
+	if f.Namespace != "" && f.Namespace != f.Deploy.Namespace && f.Deploy.Namespace != "" {
+		// TODO: when new prompt is implemented, add here a "are you sure?" check possibly
```

Comment:
Conceptually, yes, but since we are in the client, and the client is expected to be a library not a CLI, what would probably need to happen here for us to implement a confirmation prompt is:  expose a method which would allow the _CLI_ to do the check `if moving(f) { Prompt("This will remove the old one, groovy? (Y/n):") }`